### ISSUE TITLE
Enable Dark Theme

### ISSUE THREAD

**Issue by ParticleCore** [ParticleCore/Iridium#741]
Implement option that will keep dark theme always enabled even if the cookie saving the setting is deleted/expired.

**Comment by ParticleCore** [comment]
YouTube's dark theme setting is saved locally on the user's browser in a cookie.

The cookie name is `PREF` and the setting is a bit mask grouped with other settings in group `f6`.

This setting is saved in hexadecimal format, but it is used in binary.

YouTube toggles two bits instead of one whenever it saves the dark theme toggle state, one value is in position 2^10 and the other in 2^19.

````javascript

    OR = function(a, b, c) {
        a = "f" + (Math.floor(b / 31) + 1);
        b = 1 << b % 31;
        var d = nza(a) || 0;
        d = c ? d | b : d & ~b;
        0 == d ? delete JR[a] : (c = d.toString(16),
        JR[a] = c.toString())
    };

    f.handleSignalActionToggleDarkThemeOn = function() {
        OR(0, 165, !0); // 1 << 165 % 31= 1024 = 2^10
        OR(0, 174, !1); // 1 << 174 % 31= 524288 = 2^19
        this.prefs.save();
        this.actionSenderBehavior.sendAction("yt-dark-mode-toggled-action", [!0]);
        this.setLabel()
    };
    f.handleSignalActionToggleDarkThemeOff = function() {
        OR(0, 165, !1); // 1 << 165 % 31= 1024 = 2^10
        OR(0, 174, !0); // 1 << 174 % 31= 524288 = 2^19
        this.prefs.save();
        this.actionSenderBehavior.sendAction("yt-dark-mode-toggled-action", [!1]);
        this.setLabel()
    };
````

The `f6` group value must not be set to `400` directly in order to turn dark theme on in a cookie since it will override any other setting that is saved in that group, instead the value should be parsed from hexadecimal to decimal and a bit operation must be used to toggle the two values correctly in their binary positions.

````javascript
// dark theme on
// f6=c1014 as an example of having more than just the dark theme setting
(parseInt("c1014", 16) & ~524288 | 1024).toString(16) = "41414"
// f6=c1014 becomes f6=41414
// exact same operation can be done for f6=80000 -> f6=400

// dark theme off
// f6=41414 as an example of having more than just the dark theme setting
(parseInt("41414", 16) & ~1024 | 524288).toString(16) = "c1014"
// f6=41414 becomes f6=c1014
// exact same operation can be done for f6=400 -> f6=80000
````

The state can be set directly in a cookie or triggered in the webpage with the following code

````javascript
// on
document.dispatchEvent(
    new CustomEvent("yt-action", {
        bubbles: !0,
        cancelable: !1,
        composed: !0,
        detail: {
            actionName: "yt-signal-action-toggle-dark-theme-on",
            disableBroadcast: false,
            optionalAction: true,
            returnValue: []
        }
    })
);

// off
document.dispatchEvent(
    new CustomEvent("yt-action", {
        bubbles: !0,
        cancelable: !1,
        composed: !0,
        detail: {
            actionName: "yt-signal-action-toggle-dark-theme-off",
            disableBroadcast: false,
            optionalAction: true,
            returnValue: []
        }
    })
);
````

This is a toggle state meaning each one can be used to toggle the theme on and off regardless of the value in the `actionName`, however the state in the YouTube menu might get mixed up and show `on` when the theme is off and vice-versa, so it is best to use the respective `actionName` to turn the theme on and off and for that it might be needed to check the theme status before toggling it in order for it to toggle correctly.

**Comment by ParticleCore** [comment]
Implemented in 1.0.0

**Comment by sksnitj** [comment]
Hi @ParticleCore 
thanks for the details. 
Can you please explain how did you find on and off code. Thanks!
